scipy/signal/ltisys.py

```diff
@@ -790,16 +835,7 @@ def impulse(system, X0=None, T=None, N=None):
     else:
         T = asarray(T)
 
-    h = zeros(T.shape, sys.A.dtype)
-    s, v = linalg.eig(sys.A)
-    vi = linalg.inv(v)
-    C = sys.C
-    for k in range(len(h)):
-        es = diag(numpy.exp(s * T[k]))
-        eA = dot(dot(v, es), vi)
-        eA = _cast_to_array_dtype(eA, h)
-        h[k] = squeeze(dot(dot(C, eA), B))
-
+    _t, h, _x = lsim(sys, 0., T, X, interp=0)
```

Comment:
Can you make this ``interp=False`` and change the description of ``interp`` in the ``lsim`` docstring to be a ``bool``?